### DISCUSSION TITLE
code blocks with headings

### DISCUSSION THREAD

**Discussion by benkoshy** [benkoshy/pagy#7] [General]
My two cents: headings in code blocks, for the most part, do not add much value. I've tried both and I would recommend using them only on an ad-hoc basis. @ddnexus

**Comment by ddnexus** [comment]
Not sure if we are talking about the same thing: I like the standardization of the filename or logical function as header of the same block, not a separate piece. I find them more clean and atomically explain what is the block all about.

**Reply by benkoshy** [reply]
I'll standardise to this format. 


\||| pagy.rb
\```rb
\# hello world
\```
\|||

**Comment by ddnexus** [comment]
Please, to be consistent follow the same style used for the already implemented. There are a few like the pagy initializer, controller actions, view, etc. that can be standarly applied.

**Reply by benkoshy** [reply]
agreed, standardisation is key.  Example: https://benkoshy.github.io/pagy/docs/api/javascript/setup/#rails-with-assets-pipeline

![heading](https://user-images.githubusercontent.com/15097447/177074540-c83d157e-b8c1-495c-8db6-b7c6f27cbdaf.png)

The markup to produce the above:

\||| pagy.rb (initializer)
\```rb
\# etc
\```
\|||
